flytekit/deck/renderer.py

```diff
@@ -0,0 +1,22 @@
+import typing
+from abc import ABC
+
+import pandas as pd
+
+
+class Renderer(ABC):
+    def render(self) -> str:
+        """Convert a object(markdown, pandas.dataframe) to HTML and return HTML as a unicode string.
+        Returns: An HTML document as a string.
+        """
+        raise NotImplementedError
+
+
+class FrameRenderer(Renderer):
```

Comment:
After we make the RenderableProtocol, we do not need to extend it.
```suggestion
class FrameRenderer(Renderer):
```

We can create a new Renderable called
```python
class TopFrameRenderable():
    ....
```